ownCloud/Client/ClientItemCell.swift

```diff
@@ -0,0 +1,115 @@
+		prepareViewAndConstraints()
+	}
+
+	required init?(coder aDecoder: NSCoder) {
+		super.init(coder: aDecoder)
+	}
+
+	func prepareViewAndConstraints() {
+		titleLabel.translatesAutoresizingMaskIntoConstraints = false
+		detailLabel.translatesAutoresizingMaskIntoConstraints = false
+		iconView.translatesAutoresizingMaskIntoConstraints = false
+		iconView.contentMode = .scaleAspectFit
+
+		titleLabel.font = UIFont.systemFont(ofSize: 17, weight: UIFont.Weight.semibold)
+		detailLabel.font = UIFont.systemFont(ofSize: 14)
+
+		detailLabel.textColor = UIColor.gray
+
+		self.contentView.addSubview(titleLabel)
+		self.contentView.addSubview(detailLabel)
+		self.contentView.addSubview(iconView)
+
+		iconView.leftAnchor.constraint(equalTo: self.contentView.leftAnchor, constant: 20).isActive = true
+		iconView.rightAnchor.constraint(equalTo: titleLabel.leftAnchor, constant: -25).isActive = true
+		iconView.rightAnchor.constraint(equalTo: detailLabel.leftAnchor, constant: -25).isActive = true
+
+		titleLabel.rightAnchor.constraint(equalTo: self.contentView.rightAnchor, constant: -20).isActive = true
+		detailLabel.rightAnchor.constraint(equalTo: self.contentView.rightAnchor, constant: -20).isActive = true
+
+		iconView.widthAnchor.constraint(equalToConstant: 40).isActive = true
+		iconView.centerYAnchor.constraint(equalTo: self.contentView.centerYAnchor).isActive = true
+
+		titleLabel.topAnchor.constraint(equalTo: self.contentView.topAnchor, constant: 20).isActive = true
+		titleLabel.bottomAnchor.constraint(equalTo: detailLabel.topAnchor, constant: -5).isActive = true
+		detailLabel.bottomAnchor.constraint(equalTo: self.contentView.bottomAnchor, constant: -20).isActive = true
+
+		iconView.setContentHuggingPriority(UILayoutPriority.required, for: UILayoutConstraintAxis.vertical)
+		titleLabel.setContentCompressionResistancePriority(UILayoutPriority.defaultHigh, for: UILayoutConstraintAxis.vertical)
+		detailLabel.setContentCompressionResistancePriority(UILayoutPriority.defaultHigh, for: UILayoutConstraintAxis.vertical)
+	}
+
+	// MARK: - Present item
+	var _item : OCItem?
+	var item : OCItem? {
+		set(newItem) {
+			_item = newItem
+
+			if let item : OCItem = _item {
+				updateWith(item)
+			}
+		}
+
+		get {
+			return _item
+		}
+	}
+
+	func updateWith(_ item: OCItem) {
+		let iconSize : CGSize = CGSize(width: 40, height: 40)
+		var iconImage : UIImage?
+
+		iconImage = item.icon(fitInSize: iconSize)
+
+		if item.type == .collection {
+			self.detailLabel.text = "Folder"
+		} else {
+			self.detailLabel.text = item.mimeType
+		}
+
+		self.iconView.image = iconImage
+		self.titleLabel.text = item.name
+
+		if item.type == .collection {
+			self.accessoryType = .disclosureIndicator
+		} else {
+			self.accessoryType = .none
+		}
+	}
+
+	// MARK: - Themeing
+	override func applyThemeCollectionToCellContents(theme: Theme, collection: ThemeCollection) {
+		let itemState = ThemeItemState.init(selected: self.isSelected)
```

Comment:
Please provide an example of what change you'd like to see as I'm not sure I fully understand.

Passing arguments into init() to easily init & build different objects is fairly common. You can find examples for this in Apple's Swift Foundation, too:

```swift
public init(indexes: Array<Element>) {
```
[IndexPath Source](https://github.com/apple/swift-corelibs-foundation/blob/face0557631fe081765c9858abb8fb86f9795a0a/Foundation/IndexPath.swift#L565)

```swift
public init(index: Element) {
```
[IndexPath Source](https://github.com/apple/swift-corelibs-foundation/blob/face0557631fe081765c9858abb8fb86f9795a0a/Foundation/IndexPath.swift#L574)

Reply:
What I was trying to say is that there is two ways to initializate an object:
1. `let someType = SomeType.init(data: 3)` -> calling the `.init` method
2. `let someType = SomeType(data:3)` -> avoid the `.init` call. Also what apple says in the _The Swift Programming Language_ book is: 
_“If you specify a type by name, you can access the type’s initializer without using an initializer expression"_

Both ways are the same, but we have in the code the two. I think we should choose one and initialize all the objects/structs with the selected schema thinking only to have a common _coding style_

I hope this time I'm explaining myself better.

Reply:
Thanks - and sorry I didn't understand you correctly the first time. Getting rid of the .init()s sure makes for a much nicer code and I've tried to avoid using them in more recent code (old habits die slowly, though, it seems).

In the latest commit, I removed all non-super .init() calls.